plugin/ale/sign.vim

```diff
@@ -40,6 +40,34 @@ execute 'sign define ALEWarningSign text=' . g:ale_sign_warning
 \   . ' texthl=ALEWarningSign'
 sign define ALEDummySign
 
+function! ale#sign#GetStatuslineStr()
+	" Returns a formatted string that can be integrated in the 
+	" statusline
+
+	redir => signs
+	silent exec 'sign place buffer=' . bufnr('%')
+	redir END
+
+	let [errors, warnings] = [0, 0]
+	for s in split(signs, "\n")
+		if s =~# 'ALEErrorSign$'
+			let errors += 1
+		elseif s =~# 'ALEWarningSign$'
+			let warnings += 1
+		endif
+	endfor
```

Comment:
Instead of parsing the output of `sign place`, you should use the `g:ale_buffer_loclist_map` variable. It maps the buffer to the `loclist` items set up by ALE. That's less likely to lead to bugs later on, and should be easier to work with. The List is all in the standard loclist format used by Vim.